sec/insecure/insecure.go

```diff
@@ -181,31 +182,28 @@ func (ic *Conn) runHandshakeSync() error {
 }
 
 // read and write a message at the same time.
-func readWriteMsg(c msgio.ReadWriter, out *pb.Exchange) (*pb.Exchange, error) {
-	outBytes, err := out.Marshal()
-	if err != nil {
-		return nil, err
-	}
+func readWriteMsg(rw io.ReadWriter, out *pb.Exchange) (*pb.Exchange, error) {
+	const maxMsgSize = 1 << 32
```

Comment:
We can certainly reduce this to e.g. 1024kb, since we know the incoming message should be smaller than that.

Reply:
heh, good call :)